core/templates/components/state-directives/hint-editor/hint-editor.component.ts

```diff
@@ -84,10 +86,22 @@ export class HintEditorComponent implements OnInit, OnDestroy {
 
     if (contentHasChanged) {
       const hintContentId = this.hint.hintContent.contentId;
+
+      var parser = new DOMParser();
```

Comment:
Don't copy-paste this code all over the place. If you need it in multiple places, extract it into a service and call that service method instead.